code_folder/C0_data_processing.py

```diff
@@ -57,27 +57,79 @@ def simulation_settings(simulation_settings):
+                        )
+                        values_info.append(element)
+                    else:
+                        updated_values.append(element)
+                dict_asset["efficiency"]["value"] = updated_values
+                if len(values_info) > 0:
+                    dict_asset["efficiency"].update({"values_info": values_info})
+
+        # same distinction of values provided with dictionaries (one input and one output) or list (multiple).
+        # The can at turn be scalars, mantained, or timeseries
         for sector in dict_values["energyStorage"]:
             for asset in dict_values["energyStorage"][sector]:
                 for component in ["capacity", "charging_power", "discharging_power"]:
                     dict_asset = dict_values["energyStorage"][sector][asset][component]
                     for parameter in ["efficiency", "soc_min", "soc_max"]:
-                        if (
-                            parameter in dict_asset
-                            and "file_name" in dict_asset[parameter]
+                        if parameter in dict_asset and isinstance(
+                            dict_asset[parameter]["value"], dict
                         ):
                             helpers.receive_timeseries_from_csv(
                                 dict_values["simulation_settings"],
                                 dict_asset,
                                 parameter,
                             )
+                        elif parameter in dict_asset and isinstance(
+                            dict_asset[parameter]["value"], list
+                        ):
+                            updated_values = []
+                            values_info = (
+                                []
+                            )  # filenames and headers will be stored to allow keeping track of the timeseries generation
+                            for element in dict_asset[parameter]["value"]:
+                                if isinstance(element, dict):
+                                    updated_values.append(
+                                        helpers.get_timeseries_multiple_flows(
+                                            dict_values["simulation_settings"],
+                                            dict_asset,
+                                            element["file_name"],
+                                            element["header"],
+                                        )
+                                    )
+                                    values_info.append(element)
+                                else:
+                                    updated_values.append(element)
+
+                            dict_asset[parameter]["value"] = updated_values
+                            if len(values_info) > 0:
+                                dict_asset[parameter].update(
+                                    {"values_info": values_info}
+                                )
```

Comment:
Is line 107 to 132 (almost) a duplicate of line 74 to 92? Then a generalized function could be defined to shorten the code